synfig-studio/src/gui/states/state_bone.cpp

```diff
@@ -66,6 +66,19 @@ using namespace synfigapp;
 
 /* === M A C R O S ========================================================= */
 
+#ifndef LAYER_CREATION
+#define LAYER_CREATION(button, fun, stockid, tooltip)	\
+	{ \
+		Gtk::Image *icon = manage(new Gtk::Image(Gtk::StockID(stockid), \
+			Gtk::ICON_SIZE_SMALL_TOOLBAR)); \
+		button.add(*icon); \
+	} \
+	button.set_relief(Gtk::RELIEF_NONE); \
+	button.set_tooltip_text(tooltip); \
+	button.signal_toggled().connect(sigc::mem_fun(*this, \
+		&studio::StateBone_Context::fun))
+#endif
```

Comment:
I think this can be templated. For users who debug, macro functions will cause them a lot of headaches.

From [C++ Advantages/Disadvantages of Implementing functions as macros](https://stackoverflow.com/questions/16305844/c-advantages-disadvantages-of-implementing-functions-as-macros)
> The closest rule of thumb you can get is whether something needs to happen or change during compilation. If so, consider a macro, otherwise just code it up. Remember, templates count as code and can do quite a few things you might be tempted to use a macro for, but can be somewhat safer.

There is no performance overhead for this class so it's safe to template your macro function.

Reply:
Or a function since your argument data types are same.

Reply:
Even `LAYER_CREATION` as for `SPACING` is used a lot in Toolboxes and as for the latter probably the best way to deal with this separately?

Reply:
OK 👌